docs/src/main/sphinx/admin/properties-http-client.rst

```diff
@@ -0,0 +1,358 @@
+* **Default value:** ``20``
+
+Set the maximum connections allowed per server.
+
+``http-client.max-requests-queued-per-destination``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-integer`
+* **Minimum value:** ``1``
+* **Default value:** ``1024``
+
+Set the maximum number of queued connections allowed before failure.
+
+``http-client.max-content-length``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-data-size`
+* **Default value:** ``16MB``
+
+Set the maximum allowed size of the HTTP request/response.
+
+``http-client.request-buffer-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-data-size`
+* **Maximum value:** ``32MB``
+* **Default value:** ``4KB``
+
+Set the size of the buffer used for larger requests.
+
+``http-client.response-buffer-size``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-data-size`
+* **Maximum value:** ``32MB``
+* **Default value:** ``16KB``
+
+Set the size of the buffer used for larger responses.
+
+``http-client.socks-proxy``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-string`
+
+Set the host and port for a SOCKS proxy.
+
+``http-client.http-proxy``
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-string`
+
+Set the host and port for an HTTP proxy.
+
+``http-client.http-proxy.secure``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-boolean`
+* **Default value:** ``false``
+
+Enable HTTPS for the proxy.
+
+``http-client.key-store-path``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-string`
+
+Set the file path on the server to the keystore file.
+
+``http-client.key-store-password``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-string`
+
+Set the password for the keystore.
+
+``http-client.trust-store-path``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-string`
+
+Set the file path on the server to the truststore file.
+
+``http-client.trust-store-password``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-string`
+
+Set the password for the truststore.
+
+``http-client.secure-random-algorithm``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-string`
+
+Specifies which secure random algorithm is used to secure the connection. The
+default varies by operating system. Algorithms are specified according to
+standard algorithm name documentation.
+
+?Possible types include ``NativePRNG``, ``NativePRNGBlocking``,
+``NativePRNGNonBlocking``, ``PKCS11`` ``SHA1PRNG``, and ``Windows-PRNG``?
+
+``http-client.https.included-cipher``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-string`
+
+Override the default included ciphers formatted as regex.
+
+``http-client.excluded-cipher``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* **Type:** :ref:`prop-type-string`
+
+Override the default excluded ciphers formatted as regex.
+
+``http-client.automatic-shared-secret``
```

Comment:
Remove this since it's not applicable to external services and we configure this automatically for internal.